src/Chainweb/Pact/PactService.hs

```diff
@@ -796,9 +796,9 @@ execLocal cmd = withDiscardedBatch $ do
     -- necessary information for this call to return sensible values.
     parentHeader <- liftIO $! lookupM bhDb bhash
     withCheckpointer target "execLocal" $ \(PactDbEnv' pdbenv) -> do
-        PactServiceEnv{..} <- ask
         mc <- use psInitCache
-        r <- withBlockData parentHeader $
+        r <- withBlockData parentHeader $ do
+             PactServiceEnv{..} <- ask
```

Comment:
This seems like an unnecessary change -- if it is let's remove it to keep this PR focused on tests